### PR TITLE
use '0' as letter dir for funky software names that don't start with a letter

### PR DESCRIPTION
**Pull request by boegel**
I noticed this when looking into an easyconfig file to `3to2` (required for https://github.com/hpcugent/easybuild-easyconfigs/pull/3655)...

### CODE REVIEW

File: easybuild/framework/easyconfig/easyconfig.py [hunk 1]
```diff
@@ -1285,6 +1285,19 @@ def process_easyconfig(path, build_specs=None, validate=True, parse_only=False,
     return easyconfigs
 
 
+def letter_dir_for(name):
+    """
+    Determine 'letter' directory for specified software name.
+    This usually just the 1st letter of the software name (in lowercase),
+    except for funky software names, e.g. ones starting with a digit.
```
Comment (wpoely86) [inline easybuild/framework/easyconfig/easyconfig.py]: why not use number instead of something like `4real` or so?

Comment (boegel) [inline easybuild/framework/easyconfig/easyconfig.py]: well, it should be a catch-all, I don't care how it's called exactly...

I can use `_` instead?

Comment (wpoely86) [inline easybuild/framework/easyconfig/easyconfig.py]: `0` is just fine.

**Comment by boegel** [comment]
Thanks for the review @wpoely86!

**Pull request merged**